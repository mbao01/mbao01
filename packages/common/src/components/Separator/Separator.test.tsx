import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { Separator } from "./";

describe("Separator", () => {
  it("has horizontal orientation by default", () => {
    const { asFragment } = render(<Separator data-testid="separator" />);

    expect(screen.getByTestId("separator")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it.each(["horizontal", "vertical"] as const)("has %s orientation", (orientation) => {
    const { asFragment } = render(<Separator orientation={orientation} data-testid="separator" />);

    expect(screen.getByTestId("separator")).toHaveAttribute("data-orientation", orientation);
    expect(asFragment()).toMatchSnapshot();
  });

  it("is non presentational/decorative", () => {
    const { asFragment } = render(<Separator decorative={false} data-testid="separator" />);

    // separator has a role
    expect(screen.getByRole("separator")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
