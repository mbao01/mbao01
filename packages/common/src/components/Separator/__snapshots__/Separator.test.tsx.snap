// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Separator > has horizontal orientation 1`] = `
<DocumentFragment>
  <div
    class="shrink-0 bg-base-300 h-[1px] w-full"
    data-orientation="horizontal"
    data-testid="separator"
    role="none"
  />
</DocumentFragment>
`;

exports[`Separator > has horizontal orientation by default 1`] = `
<DocumentFragment>
  <div
    class="shrink-0 bg-base-300 h-[1px] w-full"
    data-orientation="horizontal"
    data-testid="separator"
    role="none"
  />
</DocumentFragment>
`;

exports[`Separator > has vertical orientation 1`] = `
<DocumentFragment>
  <div
    class="shrink-0 bg-base-300 h-full w-[1px]"
    data-orientation="vertical"
    data-testid="separator"
    role="none"
  />
</DocumentFragment>
`;

exports[`Separator > is non presentational/decorative 1`] = `
<DocumentFragment>
  <div
    class="shrink-0 bg-base-300 h-[1px] w-full"
    data-orientation="horizontal"
    data-testid="separator"
    role="separator"
  />
</DocumentFragment>
`;
